pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1336 @@
+            upperBound = ' \\leq 0 '
+        else:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+
+    elif domainName in ['NegativeReals', 'NegativeIntegers']:
+        if lowerBoundValue is not None:
+            if lowerBoundValue >= 0:
+                raise InfeasibleConstraintException(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
+            else:
+                lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ''
+
+        if upperBoundValue >= 0:
+            upperBound = ' < 0 '
+        else:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+
+    elif domainName in ['NonNegativeReals', 'NonNegativeIntegers']:
+        if lowerBoundValue > 0:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ' 0 \\leq '
+
+        if upperBoundValue is not None:
+            if upperBoundValue < 0:
+                raise InfeasibleConstraintException(
+                    'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+                )
+            elif upperBoundValue == 0:
+                upperBound = ' = 0 '
+            else:
+                upperBound = ' \\leq ' + str(upperBoundValue)
+        else:
+            upperBound = ''
+
+    elif domainName in ['Boolean', 'Binary', 'Any', 'AnyWithNone', 'EmptySet']:
+        lowerBound = ''
+        upperBound = ''
+
+    elif domainName in ['UnitInterval', 'PercentFraction']:
+        if lowerBoundValue > 1:
+            raise InfeasibleConstraintException(
+                'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+            )
+        elif lowerBoundValue == 1:
+            lowerBound = ' = 1 '
+        elif lowerBoundValue > 0:
+            lowerBound = str(lowerBoundValue) + ' \\leq '
+        else:
+            lowerBound = ' 0 \\leq '
+
+        if upperBoundValue < 0:
+            raise InfeasibleConstraintException(
+                'Formulation is infeasible due to bounds on variable %s' % (vr.name)
+            )
+        elif upperBoundValue == 0:
+            upperBound = ' = 0 '
+        elif upperBoundValue < 1:
+            upperBound = ' \\leq ' + str(upperBoundValue)
+        else:
+            upperBound = ' \\leq 1 '
+
+    else:
+        raise DeveloperError(
+            'Invalid domain somehow encountered, contact the developers'
+        )
+
+    varBoundData = {
+        'variable': vr,
+        'lowerBound': lowerBound,
+        'upperBound': upperBound,
+        'domainName': domainName,
+        'domainLatex': domainMap[domainName],
+    }
+
+    return varBoundData
+
+
+def multiple_replace(pstr, rep_dict):
+    pattern = re.compile("|".join(rep_dict.keys()), flags=re.DOTALL)
+    return pattern.sub(lambda x: rep_dict[x.group(0)], pstr)
+
+
+def latex_printer(
+    pyomo_component,
+    latex_component_map=None,
+    write_object=None,
```

Comment:
We generally refer to this as `ostream`.  It would probably be best to maintain consistency.